askama_shared/src/parser.rs

```diff
@@ -51,6 +51,44 @@ pub enum Expr<'a> {
     RustMacro(&'a str, &'a str),
 }
 
+impl Expr<'_> {
+    /// Returns `true` if enough assumptions can be made,
+    /// to determine that `self` is copyable.
+    pub fn is_copyable(&self) -> bool {
+        fn is_copyable(expr: &Expr, within_op: bool) -> bool {
```

Comment:
I don't really like the nested method here. Let's just have two methods on `Expr`, with the "inner" one called `is_copyable_within_op()` (and private)?

Reply:
Sure. I sometimes still write it like that, as in the past I encountered a few cases where it would produce more efficient binaries in the end. Unsure if that's still the case. But yeah, in this case it doesn't really matter. :)